src/gpl/src/initialPlace.cpp

```diff
@@ -106,23 +142,157 @@ void InitialPlace::doBicgstabPlace() {
 
   // set ExtId for idx reference // easy recovery
   setPlaceInstExtId();
+
+  // Parameters that don't change with iteration
+  int m = pb_->placeInsts().size(); // number of rows of matrix A
+  float tol = 1e-6; // Threshold to decide if the 
+  int reorder = 0;  // "0" for common matrix meaning no extra processing
+  int singularity = -1; // Output. Showing if the sparse matrix is signular or not. -1 = A is invertible
+
   for(int i=1; i<=ipVars_.maxIter; i++) {
     updatePinInfo();
     createSparseMatrix();
+    if (GPU == 1){
+      // Set sparse matrices from dense matrices.
+      cooRowIndexX.clear();
+      cooColIndexX.clear();
+      cooValX.clear();
+      cooRowIndexY.clear();
+      cooColIndexY.clear();
+      cooValY.clear();
+      for (size_t i = 0; i < placeInstForceMatrixX_.rows(); i ++){
+        for (size_t j = 0; j < placeInstForceMatrixX_.cols(); j ++){
+          if (placeInstForceMatrixX_.coeffRef(i,j) != 0){
```

Comment:
I'd always keep a space after a comma, so `coeffRef(i, j)` ... or with the new naming `coeffRef(row, column)`